### PR TITLE
Use WinRT VirtualKeyModifiers instead of a custom enum

### LINKED ISSUE #877: KeyChord should just use the existing WinRT types instead of defining our own
Was looking at this in conjunction with resolving suggestions for #805.  The XAML framework uses ```Windows::System::VirtualKey``` and ```Windows::System::VirtualKeyModifiers```...is there a reason that Terminal should create their own types here that are now maintaining modifiers and enum values for something that already exists?  Retrieving the keybindings with a ``KeyChord`` that has it's own values is proving to be challenging using them anywhere in the UI due to different types and not being able to cast them accordingly well.

_Originally posted by @timheuer in https://github.com/microsoft/terminal/pull/748#issuecomment-493255788_

### PR DESCRIPTION
**Pull request by mimvdb**
Replaces `KeyModifiers` with the pretty much equivalent
`VirtualKeyModifiers` enum in winrt.

After doing this I noticed #10593 which changes the KeyChords a lot, but
it seems these PRs are still compatible

The issue also mentions replacing Vkey with
`Windows::System::VirtualKey`, but I chose not to because that enum only
includes a subset of the keys terminal supports here (no VK_OEM_* keys)

## Validation Steps Performed
Changed key bind in config, and confirmed it still works after
restarting terminal

Closes #877

### CODE REVIEW

**Review by lhecker — APPROVED** [review]
This PR removes the need for `ConvertVKModifiers`. Nice. 👍
If you want to you could write
```cpp
using VirtualKeyModifiers = winrt::Windows::System::VirtualKeyModifiers;
```
here and there, to shorten the code lines, but I'm fine with merging as it is.

**Comment by DHowett** [comment]
So, our initial design for the terminal settings model was intended to avoid a dependency on the Windows.UI namespace-- it is a big namespace that we may not want consumers to depend on. 

@zadjii-msft and @carlos-zamora may have opinions.

**Comment by DHowett** [comment]
(At the same time, thanks for the contribution! It's good to have you here :))

**Comment by lhecker** [comment]
@DHowett Isn't `VirtualKeyModifiers` part of the `Windows.System` namespace?

**Comment by mimvdb** [comment]
Windows.UI is used in several places in the terminal settings model. For example, in `ActionArgs` `Windows::UI::Color` for `TabColor`. The `Windows.System` namespace does seem te be new, although `winrt/Windows.System.h` is included in the pch.

**Comment by DHowett** [comment]
> Windows.UI is used in several places in the terminal settings model. For example, in `ActionArgs` `Windows::UI::Color` for `TabColor`. The `Windows.System` namespace does seem te be new, although `winrt/Windows.System.h` is included in the pch.

> @DHowett Dustin Howett FTE Isn't `VirtualKeyModifiers` part of the `Windows.System` namespace?

Shows what I get for triaging pull requests on my phone and not reading their bodies. Thanks all! This looks fine to me, but I'd prefer @zadjii-msft sign off.

---


@msftbot make sure @zadjii-msft signs off

**Comment by DHowett** [comment]
@msftbot make sure @zadjii-msft signs off

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -1393,8 +1394,8 @@ namespace winrt::TerminalApp::implementation
             // TODO: Modify this when https://github.com/microsoft/terminal/issues/877 is resolved
             menuShortcut.Key(static_cast<Windows::System::VirtualKey>(keyChord.Vkey()));
 
-            // inspect the modifiers from the KeyChord and set the flags int he XAML value
-            auto modifiers = ActionMap::ConvertVKModifiers(keyChord.Modifiers());
+            // inspect the modifiers from the KeyChord and set the flags in the XAML value
+            auto modifiers = keyChord.Modifiers();
 
             // add the modifiers to the shortcut
             menuShortcut.Modifiers(modifiers);
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/TerminalPage.cpp:1401]:
```suggestion
            // add the modifiers to the shortcut
            menuShortcut.Modifiers(keyChord.Modifiers());
```

**Review by carlos-zamora — APPROVED** [review]
Just a minor suggestion.

I'll probably run into a few conflicts on my end with adding a key chord listener to the actions page in SUI, but that's my problem, not yours haha.

**Review by DHowett — APPROVED** [review]

**Review by zadjii-msft — APPROVED** [review]

**Pull request merged**